src/Components/v2/CCPARequest.tsx

```diff
@@ -0,0 +1,152 @@
+import React, { useState } from "react"
+import styled from "styled-components"
+import { get } from "Utils/get"
+
+interface Props {
+  user?: User
+}
+
+const FeedbackTextAreaContainer = styled(Box)`
+  max-width: 484px;
+  width: 100%;
+`
+
+const InputContainer = styled(Flex)`
+  max-width: 484px;
+  width: 100%;
+`
+
+const LoggedOutContents = ({ setName, setEmail, setText }) => {
+  return (
+    <>
+      <Sans size="3">Your message</Sans>
+      <FeedbackTextAreaContainer my={3}>
+        <TextArea
+          onChange={({ value }) => {
+            setText(value)
+          }}
+          placeholder="Your comments here"
+          required
+        />
+      </FeedbackTextAreaContainer>
+      <InputContainer mt={2} alignContent="space-between">
+        <Box mr={1} width="50%">
+          <Input
+            name="name"
+            placeholder="Your name"
+            onChange={({ currentTarget: { value } }) => {
+              setName(value)
+            }}
+            required
+          />
+        </Box>
+        <Box width="50%">
+          <Input
+            name="email"
+            placeholder="Email address"
+            onChange={({ currentTarget: { value } }) => {
+              setEmail(value)
+            }}
+            required
+          />
+        </Box>
+      </InputContainer>
+    </>
+  )
+}
+
+const LoggedInContents = ({ name, email, setText }) => {
+  return (
+    <>
+      <InputContainer mt={2} alignContent="space-between">
+        <Box mr={1} width="50%">
+          Name: {name}
+        </Box>
+        <Box width="50%">Email: {email}</Box>
+      </InputContainer>
+      <Sans size="3">Your message</Sans>
+      <FeedbackTextAreaContainer my={3}>
+        <TextArea
+          onChange={({ value }) => {
+            setText(value)
+          }}
+          placeholder="Your comments here"
+          required
+        />
+      </FeedbackTextAreaContainer>
+    </>
+  )
+}
+
+const SuccessScreen = ({ email, name, text }) => {
+  return (
+    <>
+      Successfully sent message for:
+      <Box>email: {email}</Box>
+      <Box>name: {name}</Box>
+      <Box>text: {text}</Box>
+    </>
+  )
+}
+
+export const CCPARequest: React.FC<Props> = props => {
+  const { user } = props
+  const [showModal, setShowModal] = useState(false)
+
+  const [feedbackText, setFeedbackText] = useState(null)
+  const [email, setEmail] = useState(null)
+  const [name, setName] = useState(null)
+  const [submitted, setSubmitted] = useState(false)
+
+  const userEmail = get(props, p => p.user.email)
+  const userName = get(props, p => p.user.name)
+
+  const modalContents = submitted ? (
+    <SuccessScreen
+      text={feedbackText}
+      email={userEmail || email}
+      name={userName || name}
+    />
+  ) : !user ? (
+    <LoggedOutContents
+      setName={setName.bind(this)}
+      setEmail={setEmail.bind(this)}
+      setText={setFeedbackText.bind(this)}
```

Comment:
Does this take care of proper escaping? I always get suspicious when we re-render the user's input back to them. If this isn't part of the design's confirmation screen, I'd leave it out and avoid issues of truncation, wrapping, and escaping.

Reply:
Well, the `bind` is definitely needed, as these are callbacks to set a local state to represent the user's input. The callbacks just...set that info. There's no escaping here. There is a temporary 'Success' contents in this file, just as a placeholder to get the PR up (there was no mutation ready at the time, etc.).

Happy to pair if you want to understand these callbacks better!